### PR TITLE
go.mod: Update dependencies

### PR DESCRIPTION
**Pull request by cfergeau**
I've tested that I can ssh into a podman VM on macos using this updated code, nothing more for now.

### CODE REVIEW

**Comment by cfergeau** [comment]
CI failed because our builders still use go 1.17, but the new gvisor-tap-vsock requires go 1.18
go 1.18 is available in RHEL now, go 1.17 is out of support, so imo we can do  the switch to the newer version in the builders.

**Comment by cfergeau** [comment]
/hold
Longer story in https://github.com/crc-org/crc/pull/3465 , this is blocked until we have updated go in our macos builders

**Comment by cfergeau** [comment]
Still on hold, be we are getting closer to having a usable golang 1.18. I cherry-picked the patches from https://github.com/crc-org/crc/pull/3465

**Comment by cfergeau** [comment]
CI fails with
```
cd tools && GOBIN="/home/runner/work/crc/crc/tools/bin" go install github.com/golangci/golangci-lint/cmd/golangci-lint
# honnef.co/go/tools/unused
Error: vendor/honnef.co/go/tools/unused/unused.go:419:14: obj.Origin undefined (type *types.Var has no field or method Origin)
Error: vendor/honnef.co/go/tools/unused/unused.go:421:14: obj.Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
# honnef.co/go/tools/staticcheck
Error: vendor/honnef.co/go/tools/staticcheck/lint.go:4025:29: righti.Method(i).Origin undefined (type *types.Func has no field or method Origin)
Error: vendor/honnef.co/go/tools/staticcheck/lint.go:4030:36: sel.Obj().(*types.Func).Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
make: *** [tools/tools.mk:7: /home/runner/work/crc/crc/tools/bin/golangci-lint] Error 2
Error: Process completed with exit code 2.
```
We should update to go 1.19 rather than 1.18. Since 1.20 has been released, 1.18.x is/will be unsupported soon.

**Comment by cfergeau** [comment]
/unhold
The macos builder has been updated with a golang version which avoids the regression we were seeing, so we can move forward with this change.

File: tools/go.sum [hunk 1]
```diff
@@ -1037,8 +1070,14 @@ golang.org/x/tools v0.1.8/go.mod h1:nABZi5QlRsZVlzPpHl034qft6wpY4eDcsTt5AaioBiU=
 golang.org/x/tools v0.1.9-0.20211228192929-ee1ca4ffc4da/go.mod h1:nABZi5QlRsZVlzPpHl034qft6wpY4eDcsTt5AaioBiU=
 golang.org/x/tools v0.1.9/go.mod h1:nABZi5QlRsZVlzPpHl034qft6wpY4eDcsTt5AaioBiU=
 golang.org/x/tools v0.1.10/go.mod h1:Uh6Zz+xoGYZom868N8YTex3t7RhtHDBrE8Gzo9bV56E=
+<<<<<<< HEAD
 golang.org/x/tools v0.1.11 h1:loJ25fNOEhSXfHrpoGj91eCUThwdNX6u24rO1xnNteY=
+=======
+golang.org/x/tools v0.1.11-0.20220513221640-090b14e8501f/go.mod h1:SgwaegtQh8clINPpECJMqnxLv9I09HLqnW3RMqW0CA4=
+>>>>>>> 766e0a91b (build: Update golang.org/x/net)
```
Comment (anjannath) [inline tools/go.sum]: leftover git markers in commit 312c95f5f275c7c3b6c04e2076cfcf6d16c66808 but its fixed in the next commits

**Comment by anjannath** [comment]
removing the approval to stop auto-merge in case you want to update the commit 312c95f5f275c7c3b6c04e2076cfcf6d16c66808
/approve cancel

File: tools/go.sum (hunk 1 above)
Comment (cfergeau) [reply tools/go.sum]: Good catch, thanks! I fixed this.

**Review by anjannath — APPROVED** [review]

**Pull request merged**